backend/src/main.ts

```diff
@@ -0,0 +1,47 @@
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+import {
+  SwaggerModule,
+  DocumentBuilder,
+  SwaggerDocumentOptions,
+  SwaggerCustomOptions,
+} from '@nestjs/swagger';
+import { Env } from './shared/utils/enums';
+import { ValidationPipe } from '@nestjs/common';
+import { isNil } from 'lodash';
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule);
+
+  // swagger
+  const config = new DocumentBuilder()
+    .setTitle('CV-Designer')
+    .setDescription('The CV-Designer API description')
+    .setVersion('0.1')
+    .build();
+  const options: SwaggerDocumentOptions = {
+    deepScanRoutes: true,
+  };
+  const document = SwaggerModule.createDocument(app, config, options);
+  const customOptions: SwaggerCustomOptions = {
+    explorer: true,
+    useGlobalPrefix: true,
+  };
+  SwaggerModule.setup('api', app, document, customOptions);
+
```

Comment:
How does validation work here and what do we use?

Reply:
validation is a kinda broad statement. what exactly do you mean? user validation, api endpoints validation, something else?

Reply:
I meant user validation, again I didn't catch proper lines in my comment, sorry

Reply:
there is no user validation at this time, if we decide to add it, it will be done at a service level (AuthService - AuthController etc)